src/hooks/screen/useChatScreenHook.ts

```diff
@@ -27,35 +30,55 @@ interface ScrollContextProps {
 export const ScrollContext = React.createContext<ScrollContextProps | null>(null);
 
 function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
-  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {localDb, refresh, otherListener, refreshWithId, chat} = useLocalDatabaseHook();
   const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook(type);
   const {anonProfileId} = useUserAuthHook();
+  const {queue} = useDatabaseQueueHook();
+
+  const isInitiatingChatRef = React.useRef(false);
 
   const [selfAnonUserInfo, setSelfAnonUserInfo] = React.useState<any>(null);
   const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
   const initChatData = async () => {
     if (!localDb || !selectedChannel) return;
     try {
-      const myUserId = await getUserId();
-      const myAnonymousId = await getAnonymousUserId();
-      const data = (await ChatSchema.getAll(
-        localDb,
-        selectedChannel?.id,
-        myUserId,
-        myAnonymousId
-      )) as ChatSchema[];
-      setChats(data);
+      isInitiatingChatRef.current = true;
+      const myUserId = getUserId();
+      const myAnonymousId = getAnonymousUserId();
+      queue.addPriorityJob({
+        priority: QueueJobPriority.HIGH,
+        label: `get-all-chat-${selectedChannel?.name}`,
+        task: async () => {
+          const data = (await ChatSchema.getAll(
+            localDb,
+            selectedChannel?.id,
+            myUserId,
+            myAnonymousId
+          )) as ChatSchema[];
+          setChats(data);
+          isInitiatingChatRef.current = false;
+
+          return data;
+        }
+      });
 
       if (type === 'ANONYMOUS') {
-        const userInfo = await UserSchema.getSelfAnonUserInfo(
-          localDb,
-          anonProfileId,
-          selectedChannel?.id
-        );
-
-        setSelfAnonUserInfo(userInfo);
+        queue.addPriorityJob({
+          priority: QueueJobPriority.MEDIUM,
+          label: `get-anon-user-info-${selectedChannel?.name}`,
+          task: async () => {
+            const userInfo = await UserSchema.getSelfAnonUserInfo(
+              localDb,
+              anonProfileId,
+              selectedChannel?.id
+            );
+            setSelfAnonUserInfo(userInfo);
+          }
+        });
       }
     } catch (e) {
+      isInitiatingChatRef.current = false;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatData` function has been updated to use a priority queue system for fetching chat data and user information. This is a good approach as it allows prioritizing certain tasks over others, which can be beneficial in scenarios where network resources are limited or when you want to prioritize user-facing tasks over background tasks.

However, there's a potential issue with error handling. In the catch block, the error is logged but not handled further. It would be better to provide some feedback to the user or at least report this error to an error tracking service if one is in use. 

```diff
     } catch (e) {
      isInitiatingChatRef.current = false;
+       SimpleToast.show('Error fetching chat data');
+       // Consider reporting this error to an error tracking service
       console.log(e, 'error get all chat');
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->